src/StateManager.cpp

```diff
@@ -0,0 +1,16 @@
+#include "StateManager.hpp"
+
+StateManager::StateManager() {
+}
+
+void StateManager::update() {
+    this->currentState.update();
+}
+
+void StateManager::draw() {
+    currentState.draw();
```

Comment:
I meant adding `this->` everywhere, including here.
```suggestion
    this->currentState.draw();
```